### ISSUE TITLE
The xlsx-import Command Line Interface

### ISSUE THREAD

**Issue by Siemienik** [Siemienik/XToolset#10]
# The `xlsx-import-cli` package

We have cli for renderer, now we should add cli for the import. I blocked this issue earlier, because I was doubts about setting mappers in `json` config file - however it isn't a problem if we use `js` files.

## Usage:

```shell-scripts
# basic
cat invoice.xlsx > sxi config.js > result.json
sxi config.js invoice.xlsx > result.json
sxi -o result.json config.js invoice.xlsx 

#more advanced (todo)
cat invoice.xlsx > sxi config.js | sxr template.xlsx > refreshed-invoice.xlsx
```

## Requirements

* Added into ./packages/xlsx-import-cli
* NPM package name: sxi (reserved on npm)
* The interface: `sxi [-o <output-file.json>] <import-config.js> [input-file.xlsx]`. The `input-file.xlsx` isnt required if comes in sdtin.
* Should be able to pipe with `sxr` (xlsx-renderer-cli).
* Please add into main Readme.md
* Please to make documentation

## Additional info:

* May been inspired by `xlsx-renderer-cli`
* It will be nice to able cover it in 100% percent. 

## After

* Create project `xlsx-import-cli` for handling tasks targeting to it.
* Create an issue for adding TS files support [(as suggested @Metastasis)](https://github.com/Siemienik/XToolset/issues/10#issuecomment-741878219)
* Create an issue for adding support for JSON files  [(as suggested @Metastasis)](https://github.com/Siemienik/XToolset/issues/10#issuecomment-741878219)

**Comment by Metastasis** [comment]
@Siemienik 
Could you describe a little bit this issue?

1. What args should be?
2. Is this supposed to be implemented via "bin" in package.json?
3. Do you have any preferences over CLI framework or it could be any? I prefer [cac](https://www.npmjs.com/package/cac). I think it simple enough and will suite our needs

**Comment by Siemienik** [comment]
I'm going to describe it for todo in next week (after version`2.3.2`published).
It will be similar to: https://github.com/Siemienik/xlsx-renderer/issues/25 done in https://github.com/Siemienik/xlsx-renderer-cli/pull/1 (it's stuck because commander isn't test friendly :/) 

If are you looking for something todo, may take care about finishing https://github.com/Siemienik/xlsx-renderer-cli/pull/1 ? 

> What args should be? 

not described yet, however I have some ideas

> Is this supposed to be implemented via "bin" in package.json?

I consider about one of two way, 
* monorepo
* external repo `xlsx-import-cli`

in NPM I have reserved package `sxi` for this tool.

> Do you have any preferences over CLI framework or it could be any? I prefer cac. I think it simple enough and will suite our needs

No, however I want to keep 100% test coverage - it will be nice to used tool allow testing without full pack of pain. (what I investigated, https://github.com/oclif/oclif is test friendly)

**Comment by Siemienik** [comment]
```
# basic
cat invoice.xlsx > sxi config.json > result.json
sxi config.json invoice.xlsx > result.json
sxi -o result.json config.json invoice.xlsx 

#more advanced (todo)
cat invoice.xlsx > sxi config.json | sxr template.xlsx > refreshed-invoice.xlsx
```

**DRAFT reason:** _I've just realized that, config may contains value mapper that do not works in a `json` file. Have to invastigate / write a prototyp._

**Comment by Siemienik** [comment]
> **DRAFT reason:** _I've just realized that, config may contains value mapper that do not works in a `json` file. Have to invastigate / write a prototyp._

This should not be a problem after #31

**Comment by Siemienik** [comment]
I change it from Draft to Proposal, because to mamy thing should happen before we may start doing it.

However this issue is quite important imo.

**Comment by Siemienik** [comment]
> DRAFT reason: I've just realized that, config may contains value mapper that do not works in a json file. Have to invastigate / write a prototyp.

This will not be a problem when `import-cfg.json` will be replaced by JS file : `import-cfg.js` wich exports default config.

**Comment by Siemienik** [comment]
@Metastasis WDYT? should I describe something more for this issue?

**Comment by Metastasis** [comment]
@Siemienik In examples folder we used configs declared both in ts and js files. Does CLI have to support ts files?

**Comment by Siemienik** [comment]
>Also what about json files? We just ignore them for now?

I think json files will works like js one but without mappers. (json is much more secure than js file. From the other side, js file has possibility to make config dynamically). I consider about json file supporting move to new issue.

> Does CLI have to support ts files?

Imo, this should be another one issue and will analyse it separately.

WDYT?

**Comment by Metastasis** [comment]
@Siemienik Ok, sounds good. Then rest part of the issue I think is clear

**Comment by Siemienik** [comment]
Xlsx-import-cli `v0.0.1-alpha1` released! Feel welcome to test alpha version 🚀
